From 542e65deaf834c3d58c44064206445e4d18f6daa Mon Sep 17 00:00:00 2001
From: BALATON Zoltan <balaton@eik.bme.hu>
Date: Sat, 11 Jun 2011 10:50:16 +0000
Subject: Fix regression from pciaccess conversion (bug#32682)

Due to a typo in TsengGetFbAddress it is unable to get framebuffer base
address and thus fails to load with an error message saying "No valid
Framebuffer address in PCI config space" although previously it does
print the correct framebuffer address.

Signed-off-by: Julien Cristau <jcristau@debian.org>
---
diff --git a/src/tseng_driver.c b/src/tseng_driver.c
index 6992671..d36b62c 100644
--- a/src/tseng_driver.c
+++ b/src/tseng_driver.c
@@ -883,7 +883,7 @@ TsengGetFbAddress(ScrnInfoPtr pScrn)
     PDEBUG("	TsengGetFbAddress\n");
 
     /* base0 is the framebuffer and base1 is the PCI IO space. */
-    if (PCI_REGION_BASE(pTseng->PciInfo, 0, REGION_MEM)) {
+    if (!PCI_REGION_BASE(pTseng->PciInfo, 0, REGION_MEM)) {
         xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
                    "No valid Framebuffer address in PCI config space;\n");
         return FALSE;
--
cgit v0.9.0.2-2-gbebe
